chain/network/src/peer/peer_actor.rs

```diff
@@ -1087,8 +1083,19 @@ impl actix::Handler<stream::Frame> for PeerActor {
                 PeerStatus::Connecting(ConnectingStatus::Outbound { handshake_spec, .. }),
                 PeerMessage::LastEdge(edge),
             ) => {
-                // Disconnect if neighbor proposed an invalid edge.
-                if !edge.verify() {
+                // Check that the edge provided is:
+                let ok =
+                    // - for the relevant pair of peers
+                    edge.key()==&Edge::make_key(self.my_node_info.id.clone(),handshake_spec.peer_id.clone()) &&
+                    // - is not older than what we proposed originally
+                    edge.nonce() >= handshake_spec.partial_edge_info.nonce &&
```

Comment:
:thinking: should this be `>` perhaps? Looking at nonces seems like a more straightforward way to verify that this isn't our own edge echoed back, this should allow us to get rid of the signature checks below.

Reply:
I guess we might want to have `=` in some cases perhaps, but its probably worth to commet.

Reply:
done